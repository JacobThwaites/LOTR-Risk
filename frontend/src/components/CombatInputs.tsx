import React, { Dispatch, SetStateAction } from "react";
import UnitInput from "./common/UnitInput";

type Props = {
  maxAttackingDice: number,
  attackingDice: number,
  maxDefendingDice: number,
  defendingDice: number
  setAttackingDice: Dispatch<SetStateAction<number>>,
  setDefendingDice: Dispatch<SetStateAction<number>>,
  areInputsDisabled: boolean
}

export default function CombatInputs(props: Props) {
  return (
    <div className="combat-handler__dice-input">
      <div className="combat-handler__dice-input--attacking">
        <UnitInput
          label="Attacking Dice"
          name="attackingDice"
          max={props.maxAttackingDice}
          value={props.attackingDice}
          onChange={(num: number) => props.setAttackingDice(num)}
          disabled={!props.areInputsDisabled}
        />
      </div>
      <div className="combat-handler__dice-input--defending">
        <UnitInput
          label="Defending Dice"
          name="defendingDice"
          max={props.maxDefendingDice}
          value={props.defendingDice}
          onChange={(num: number) => props.setDefendingDice(num)}
          disabled={!props.areInputsDisabled}
        />
      </div>
    </div>
  );
}