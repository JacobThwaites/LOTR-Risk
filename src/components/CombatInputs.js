import React from "react";
import UnitInput from "./common/UnitInput";

function CombatInputs(props) {
  return (
    <div className="combat-handler__dice-input">
      <div className="combat-handler__dice-input--attacking">
        <UnitInput
          label="Attacking Dice"
          name="attackingDice"
          max={props.maxAttackingDice}
          value={props.attackingDice}
          onChange={props.onNumberSelect}
        />
      </div>
      <div className="combat-handler__dice-input--defending">
        <UnitInput
          label="Defending Dice"
          name="defendingDice"
          max={props.maxDefendingDice}
          value={props.defendingDice}
          onChange={props.onNumberSelect}
        />
      </div>
    </div>
  );
}

export default CombatInputs;
