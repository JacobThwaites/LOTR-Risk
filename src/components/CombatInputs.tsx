import React from "react";
import UnitInput from "./common/UnitInput";

type Props = {
  maxAttackingDice: number,
  attackingDice: number,
  onNumberSelect: Function,
  maxDefendingDice: number,
  defendingDice: number
}

export default function CombatInputs(props: Props) {
  return (
    <div className="combat-handler__dice-input">
      <div className="combat-handler__dice-input--attacking">
        <UnitInput
          label="Attacking Dice"
          name="attackingDice"
          max={props.maxAttackingDice}
          value={props.attackingDice}
          onChange={props.onNumberSelect}
        />
      </div>
      <div className="combat-handler__dice-input--defending">
        <UnitInput
          label="Defending Dice"
          name="defendingDice"
          max={props.maxDefendingDice}
          value={props.defendingDice}
          onChange={props.onNumberSelect}
        />
      </div>
    </div>
  );
}