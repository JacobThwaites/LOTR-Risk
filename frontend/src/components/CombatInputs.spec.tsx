import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import React from 'react';
import CombatInputs from './CombatInputs';

it('renders CombatInputs correctly', async () => {
    const maxAttackingDice = 3;
    let attackingDice = 2;
    const setAttackingDice = jest.fn((input) => (attackingDice = input));
    const areInputsDisabled = false;

    const { container } = render(
        <CombatInputs 
            maxAttackingDice={maxAttackingDice}
            attackingDice={attackingDice}
            setAttackingDice={setAttackingDice}
            areInputsDisabled={areInputsDisabled}
        />
    );

    expect(container).toMatchSnapshot();
});
