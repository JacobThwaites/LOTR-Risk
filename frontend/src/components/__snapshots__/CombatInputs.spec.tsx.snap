// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CombatInputs correctly 1`] = `
<div>
  <div
    class="combat-handler__dice-input"
  >
    <label>
      Attacking Dice: 
    </label>
    <span
      class="react-numeric-input"
      style="position: relative; display: inline-block;"
    >
      <input
        class="number-selector"
        disabled=""
        name="attackingDice"
        pattern=".*"
        style="padding-right: 3ex; box-sizing: border-box; border: 1px solid #ccc; border-radius: 2px; padding-left: 4px; display: block; line-height: normal; color: rgba(0, 0, 0, 0.3); text-shadow: 0 1px 0 rgba(255, 255, 255, 0.8);"
        type="text"
        value="2"
      />
      <b
        style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0,0,0,.1); border-style: solid; text-align: center; cursor: not-allowed; transition: all 0.1s; background: rgba(0, 0, 0, 0.1); box-shadow: none; top: 2px; bottom: 50%; border-radius: 2px 2px 0 0; border-width: 1px 1px 0px 1px; opacity: 0.5;"
      >
        <i
          style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0px 0.6ex 0.6ex 0.6ex; border-color: transparent transparent rgba(0, 0, 0, 0.7); border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
        />
      </b>
      <b
        style="position: absolute; right: 2px; width: 2.26ex; border-color: rgba(0,0,0,.1); border-style: solid; text-align: center; cursor: not-allowed; transition: all 0.1s; background: rgba(0, 0, 0, 0.1); box-shadow: none; top: 50%; bottom: 2px; border-radius: 0 0 2px 2px; border-width: 0px 1px 1px 1px; opacity: 0.5;"
      >
        <i
          style="position: absolute; top: 50%; left: 50%; width: 0px; height: 0px; border-width: 0.6ex 0.6ex 0px 0.6ex; border-color: rgba(0, 0, 0, 0.7) transparent transparent; border-style: solid; margin: -0.3ex 0px 0px -0.56ex;"
        />
      </b>
    </span>
  </div>
</div>
`;
